### ISSUE TITLE
PyPy is ignored

### ISSUE THREAD

**Issue by adiroiban** [ryanhiebert/tox-travis#16]
Here is my attempt to enable pypy via tox-travis 

https://github.com/twisted/twisted/pull/226

In the first commit I just added pypy, but it was not updated in the env list.

BTW. Why is tox-travis not running its own tests via tox-travis ? :)

**Comment by jayvdb** [comment]
As far as I know, pypy and pypy3 work correctly with tox-travis. 

It is only using the travis supplied pypy/pypy3 , but it works.

Adding a modern pypy to travis is the subject of at least
* https://github.com/travis-ci/travis-ci/issues/3202
* https://github.com/travis-ci/travis-ci/issues/4756
* https://github.com/travis-ci/travis-ci/issues/5027
(and probably others)
Those issues have posts including a pyenv based recipe much like your own.

Are you suggesting that tox-travis should fill the void and somehow add extra pypy versions?
Or does tox-travis need to do something to support your pyenv recipe?

**Comment by ryanhiebert** [comment]
Yes, I'd appreciate it if you clarified your intent. My purpose so far with the PyPy support is to simply have it run with the version of pypy (and pypy3) that Travis supports. If simpler support for more or updated of PyPy than the default Travis ones is the goal, that seems like an admirable one to me, and one that I'd support.

Also, `tox-travis` is most certainly running it's own tests using `tox-travis`. That's the only way to really test it. The install line is this magic bit, which installs the version under question, rather than an arbitrary one from PyPI, so that I can properly test it, including the actual behavior on Travis.

https://github.com/ryanhiebert/tox-travis/blob/master/.travis.yml#L11

```yaml
install: pip install .
```

Finally, holy smokes: Twisted is using my project! That's cool beyond words!

**Comment by ryanhiebert** [comment]
TL;DR: You needed to add `pypy` to the list of tested versions in the `.travis.yml` file.

Looking back some more, I can see that you're clear that it didn't run the `pypy` environment, even though you added it to the `.travis.yml`. It appears that you didn't add `pypy` to the matrix of versions in `.travis.yml`. It's important to remember that This is a _tox_ plugin, that helps it know about Travis, but it doesn't give any clues to Travis about tox.

Ultimately, every part of your build matrix _must_ be specified in the `.travis.yml`, and I'm just trying to remove the parts that are redundant between the two files so you don't need to repeat yourself for a simple configuration.

**Comment by rodrigc** [comment]
@ryanhiebert I submitted the patch for Twisted to use tox-travis: https://github.com/twisted/twisted/commit/a241f4201fbfb9d1afdeebb6522c47d5b2dbb713
Thanks for writing the software.  It saved me a lot of time.

**Comment by adiroiban** [comment]
True. Thanks.  It was my fault. Sorry for the trouble.

**Comment by jayvdb** [comment]
It could be possible to check that the `.travis.yml` matrix has all of the versions that are in `tox.ini` `[tox:travis]`, as a quick run-time check and warning that the two files have been set up correctly / they havent been shoved out of sync accidentally.

> Ultimately, every part of your build matrix must be specified in the .travis.yml

Worth thinking about whether this is avoidable, at least for simple cases, and it looks like it is.

If `language: python` is specified in `.travis.yml`, and it doesnt specify any `python: xx` versions, Python 2.7 will be used, and most of the tox versions are available.  The following will do what most people expect in the simplest of cases for a simple single job:

```yaml
[tox:travis]
2.7 = py27,py32,py33,py34,pypy,pypy3
```

https://travis-ci.org/jayvdb/my-ci-test/builds/140569757

Obviously that doesnt use the full power of tox-travis, but maybe it is useful to some people, most notably people who want only a single Travis job due to increased execution time of using a separate travis job for each Python version, but wish to take advantage of other features of tox-travis (even if it is only the ability to change the travis job from tox.ini, without changing the default envlist.

It would be nice if the following syntax also worked for this simple scenario, as it is the same syntax as `[tox:jenkins]`:

```yaml
[tox:travis]
envlist = py27,py32,py33,py34,pypy,pypy3
```

**Comment by ryanhiebert** [comment]
@jayvdb: Thanks for the continuing discussion. I appreciate having other folks thinking about this too. I've opened up #18 and #19 to continue these conversations. They are somewhat related, but seem to be separate issues to me.
